dashboard/config/routes.rb

```diff
@@ -9,7 +9,7 @@
     get '/weblab/footer', to: 'projects#weblab_footer'
   end
 
-  constraints host: /.*code.org.*|.*hourofcode.com.*/ do
+  constraints host: /^((?!#{CDO.codeprojects_hostname}).)*$/ do
```

Comment:
awesome! glad to see this as I know this has tripped some other folks up lately.

TBH my regex-fu is not strong enough to understand exactly what this is doing (why is the trailing dot and outer parens needed?), so would vote for a brief comment explaining that this is everything-but-codeprojects

Reply:
will do! I tested removing some of the pieces but before I used `CDO.codeprojects_hostname` -- so I can now clean this up a lot.